provider/resources.go

```diff
@@ -49,37 +50,60 @@ func makeResource(mod string, res string) tokens.Type {
 
 type computeIDFunc = func(ctx context.Context, state resource.PropertyMap) (resource.ID, error)
 
-func computeIDField(field resource.PropertyKey) computeIDFunc {
-	return func(ctx context.Context, state resource.PropertyMap) (resource.ID, error) {
-		fieldValue, ok := state[field]
-		if !ok {
-			return "", fmt.Errorf("id: could not find required property '%s'", field)
-		}
+func delegateIDField(field resource.PropertyKey) computeIDFunc {
+	return tfbridge.DelegateIDField(field, "meraki", "https://github.com/pulumi/pulumi-meraki")
+}
+
+func delegateIDProperty(field resource.PropertyPath) computeIDFunc {
+	return tfbridge.DelegateIDProperty(field, "meraki", "https://github.com/pulumi/pulumi-meraki")
+}
 
-		// ComputeID is only called during when preview=false, so we don't need to
-		// deal with computed properties.
+func applyResourceIDs(info tfbridge.PropertyVisitInfo) (tfbridge.PropertyVisitResult, error) {
+	path := info.SchemaPath()
 
-		if fieldValue.IsSecret() || (fieldValue.IsOutput() && fieldValue.OutputValue().Secret) {
-			msg := fmt.Sprintf("Setting non-secret resource ID as '%s' (which is secret)", field)
-			tfbridge.GetLogger(ctx).Warn(msg)
-			if fieldValue.IsSecret() {
-				fieldValue = fieldValue.SecretValue().Element
-			} else {
-				fieldValue = fieldValue.OutputValue().Element
+	res, ok := info.Root.(tfbridge.VisitResourceRoot)
+
+	if !ok || len(path) > 1 || (res.Info != nil && res.Info.ComputeID != nil) {
+		return tfbridge.PropertyVisitResult{HasEffect: false}, nil
+	}
+
+	// If the resource already has an ID property, just use that.
+	if _, ok := res.Schema.Schema().GetOk("id"); ok {
+		return tfbridge.PropertyVisitResult{HasEffect: false}, nil
+	}
+
+	// On each resource, we check for 1 of the 4 known ID properties.
+	//
+	// If a resource has just one, we set that as the ID. If a resource has more then
+	// 1, we error and allow an operator to pick which one.
```

Comment:
nit. I don't think we have to be quite this ambivalent about the correct id; the name prefixes seem to correspond with the id name pretty strictly, with the caveat that the id field is nested for ` meraki_administered_licensing_subscription_subscriptions_claim` and `meraki_administered_licensing_subscription_subscriptions_claim_key_validate`

Reply:
Good point. I addressed this is [a9d87a4](https://github.com/pulumi/pulumi-meraki/pull/5/commits/a9d87a41766687c95762e63f69858af3580d7af5).